Sources/Networking/NetworkService/AFSessionManager/AFSessionManager.swift

```diff
@@ -4,16 +4,24 @@ import NetworkInterface
 
 open class AFSessionManager: Session {
     
-    public static func `default`(adaptHeaders: ((inout [String: String]) -> Void)?) -> AFSessionManager {
-        var interceptor: Interceptor?
+    public static func `default`(adaptHeaders: ((inout [String: String]) -> Void)?,
+                                 retryProvider: RetryProviderProtocol? = nil,
+                                 maxRetryCount: Int = 3) -> AFSessionManager {
         var adapter: HeadersAdapter?
+        var retrier: RetrayablePolicy?
+        var interceptor: Interceptor?
         
         if let adaptHeaders = adaptHeaders {
```

Comment:
Seems to be broken logic in this bock, can't we have an interceptor without headers? We can...

I suggest it should be

let adapter = flatMap...
let retires = flatMap...

interceptor = Interceptor(adapter: adapter, retrier: retrier)